### ISSUE TITLE
Error: primary pod is not in Ready state, when do a full backup.

### ISSUE THREAD

**Issue by satriashp** [CrunchyData/postgres-operator#2512]
## Overview

I can't do a full backup on my cluster (name: `dwh`). 
Output message say `Error: primary pod is not in Ready state`

## Environment

Please provide the following details:

- Platform: `Kubernetes`
- Platform Version: `1.15.12`
- PGO Image Tag: `centos8-4.5.1`
- Postgres Version `11`
- Storage: `kubernetes.io/aws-ebs`

### REPRO

Provide steps to get to the error condition:

1. Run `pgo backup dwh --backup-opts="--type full --repo1-retention-full=3"`
2. Do test on a cluster `pgo test dwh` and my cluster looks fine, all test passed

### EXPECTED

1. Output message should be like this `created Pgtask backrest-backup-dwh`
2. Successfully do a full backup

### ACTUAL

1. Output message say `Error: primary pod is not in Ready state`
2. And backup process not running

## Logs

Please provided appropriate log output or any configuration files that may help troubleshoot the issue. **DO NOT** include sensitive information, such as passwords.

## Additional Information

1. Output from `pgo test dwh`
```
cluster : dwh
	Services
		primary (100.70.95.40:5432): UP
		pgbouncer (100.67.28.14:5432): UP
		replica (100.70.95.191:5432): UP
	Instances
		primary (dwh-pdkl-d9f9f5877-kfcmq): UP
		replica (dwh-tzik-84bf7c8f6f-smkd6): UP

```

**Comment by andribiz** [comment]
If anyone encounter this problem

change label of your primary node

`
kubectl label --overwrite pod your-primary-pod service-name=your-cluster
`

the solution for the problem above 

`
kubectl label --overwrite pod dwh-pdkl-d9f9f5877-kfcmq service-name= dwh
`

it seem pgo/pgbackrest look into pod label to select primary node

**Comment by adthonb** [comment]
@jkatz This is a bug. If the primary pod on the initial PostgreSQL cluster is down, the new primary pod will have a name with a cluster prefix. I have the same problem on v4.6.0 that PGO client can't find a new primary. 

1. Run `pgo backup his --backup-opts="--type=full --repo1-retention-full=2"` and get an error
`Error: primary pod is not in Ready state`

2. Run `pgo test his`
```
cluster : his
        Services
                primary (10.99.168.113:5432): UP
                replica (10.100.248.59:5432): UP
        Instances
                primary (his-brza-85bfc78766-5qcfg): UP
                replica (his-ptcw-78cf6fb745-jd26p): UP
                replica (his-wuzc-5b7ffc89dd-kr2kk): UP
```

3.  Run `kubectl get pod/his-brza-85bfc78766-5qcfg -o yaml` and see label
```
  labels:
    crunchy-pgha-scope: his
    deployment-name: his-brza
    name: his-replica
    pg-cluster: his
    pg-pod-anti-affinity: preferred
    pgo-pg-database: "true"
    pgo-version: 4.6.0
    pod-template-hash: 85bfc78766
    replica-name: his-brza
    role: master
    service-name: his-replica
    vendor: crunchydata
    workflowid: 1ead7fc7-b8a2-4111-adf1-db518cbb61c8
```

Some labels make a conflict with PGO client

**Comment by jkatz** [comment]
This should have been fixed in [4.6.1](https://access.crunchydata.com/documentation/postgres-operator/4.6.4/releases/4.6.1/):

```
- Only attempts to start scheduled backups in running pgBackRest repository Pods.
- Fix issue with pgo backup where it was unable to take a backup from a new primary after pgo failover was called.
```

The failover having similar issues to what is described.

I suggest running the latest version of v4.6.x and seeing if the issue persists. In general, it's a good idea to report bugs against the latest version of a minor release in case they were already fixed.

**Comment by benjaminjb** [comment]
Does this problem persist when running v4.6.1 (or other updated PGO version)?
